panic caused by insertChain

panic when sync:

```text
INFO [11-12|04:26:11.039] Imported new chain segment               blocks=1   txs=4    mgas=0.090   elapsed=6.618ms       mgasps=13.566  number=95,919,842 hash=58b4ab..7a48e8 dirty=0.00B
INFO [11-12|04:26:11.185] [voteHandler] Vote pool threashold reached: true, number of items in the pool: 73
INFO [11-12|04:26:11.185] Successfully commit and confirm block from continuous 3 blocks num=95,919,840 round=15777927   hash=0649ca..86c3a5
INFO [11-12|04:26:11.185] [setNewRound] new round and reset pools and workers round=15777930
INFO [11-12|04:26:11.185] Successfully processed the vote and produced QC! QcRound=15777929 QcNumOfSig=73  QcHash=58b4ab..7a48e8 QcNumber=95,919,842
INFO [11-12|04:26:11.185] [voteHandler] time cost from receive first vote under QC create elapsed=163.804876ms
INFO [11-12|04:26:21.246] Timeout pool threashold reached: true, number of items in the pool: 73
INFO [11-12|04:26:21.246] [setNewRound] new round and reset pools and workers round=15777931
INFO [11-12|04:26:21.246] [onTimeoutPoolThresholdReached] Successfully processed the timeout message and produced TC! TcRound=15777930 NumberOfTcSig=73
INFO [11-12|04:26:21.708] [voteHandler] set vote pool time         round=15777931
INFO [11-12|04:26:22.991] [voteHandler] Vote pool threashold reached: true, number of items in the pool: 73
INFO [11-12|04:26:22.991] [voteHandler] The proposed block from vote message does not exist yet, wait for the next vote to try again blockNum=95,919,843 Hash=c9aa13..71c15f Round=15777931
INFO [11-12|04:26:23.054] [voteHandler] Vote pool threashold reached: true, number of items in the pool: 74
INFO [11-12|04:26:23.054] [voteHandler] The proposed block from vote message does not exist yet, wait for the next vote to try again blockNum=95,919,843 Hash=c9aa13..71c15f Round=15777931
INFO [11-12|04:26:23.333] Imported new chain segment               blocks=1   txs=2    mgas=3.546   elapsed=697.293ms     mgasps=5.085   number=95,919,843 hash=a783d4..858432 dirty=0.00B
panic: runtime error: index out of range [0] with length 0

goroutine 202491573 [running]:
github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).insertChain(0x16252f4?, {0x0?, 0xc0e50f5a40?, 0x6?}, 0x6?)
        /home/me/XDPoSChain/core/blockchain.go:1687 +0x3448
github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).getResultBlock(0xc000004c08, 0xc0c5479d40, 0x19?)
        /home/me/XDPoSChain/core/blockchain.go:2168 +0x1d39
github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).insertBlock(0xc000004c08, 0xc0c5479d40)
        /home/me/XDPoSChain/core/blockchain.go:2243 +0x2e8
github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).InsertBlock(0xc000004c08, 0x300c00a9edc08?)
        /home/me/XDPoSChain/core/blockchain.go:2075 +0x18
github.com/XinFinOrg/XDPoSChain/eth.NewProtocolManager.func8(0xc0002e3260?)
        /home/me/XDPoSChain/eth/handler.go:228 +0x296
github.com/XinFinOrg/XDPoSChain/eth/fetcher.(*Fetcher).insert.func1()
        /home/me/XDPoSChain/eth/fetcher/fetcher.go:729 +0x1478
created by github.com/XinFinOrg/XDPoSChain/eth/fetcher.(*Fetcher).insert in goroutine 111
        /home/me/XDPoSChain/eth/fetcher/fetcher.go:671 +0x3cc
```

--- a/core/blockchain.go
+++ b/core/blockchain.go
@@ -2158,6 +2158,10 @@ func (bc *BlockChain) getResultBlock(block *types.Block, verifiedM2 bool) (*Resu
 			winner = append(winner, parent)
 			parent = bc.GetBlock(parent.ParentHash(), parent.NumberU64()-1)
 		}
+		// fix issue #1765, return at once if winner is empty
+		if len(winner) == 0 {
+			return nil, errors.New("winner is empty")
+		}
 		for j := 0; j < len(winner)/2; j++ {
 			winner[j], winner[len(winner)-1-j] = winner[len(winner)-1-j], winner[j]
 		}

